tags.go

```diff
@@ -26,10 +26,16 @@ func DefaultTagSuffix(ref, suffix string) []string {
 
 // DefaultTags returns a set of default suggested tags based on
 // the commit ref.
-func DefaultTags(ref string) []string {
+func DefaultTags(ref, commitBranch, defaultBranch string) []string {
+
+	if defaultBranch != "" && commitBranch != defaultBranch && !strings.HasPrefix(ref, "refs/tags/") {
+		return []string{}
```

Comment:
I am not sure we ever want to return an empty string here. We should always return a default tag.

Reply:
maybe we can add a new flag to confirm that if the user wants to build tags from another branch which is not equal the default branch.

1. default branch equal to commit branch. return `latest` tag
2. new flag `BuildFromBranch`
  2.1: default branch not equal to commit branch. return `commit branch` tag if BuildFromBranch is `true`
  2.2: default branch not equal to commit branch. return `empty` tag if BuildFromBranch is `false`

Reply:
> default branch not equal to commit branch. return empty tag

An empty tag is not valid and will fail

Reply:
@bradrydzewski I meant empty value like `[]string{}`

Reply:
and we can check the tag array in `Exec` like

```go
	if len(p.Build.Tags) == 0 {
		fmt.Println("build tags not found")
		return nil
	}
```

Reply:
@bradrydzewski Any other suggestion?